src/js/api/brand.js

```diff
@@ -147,7 +147,20 @@ ripe.Ripe.prototype.get3dSceneEnvironmentUrl = function(options) {
  * @returns {String} The URL of the specified initials texture.
  */
 ripe.Ripe.prototype.getTextureMapUrl = function(map, options) {
-    return null;
+    options = this._getInitials3dOptions(options);
+
+    const textureMap = new Map([
+        ["pattern", options.baseTexture],
+        ["displacement", options.displacementTexture],
+        ["metallic", options.metallicTexture],
+        ["normal", options.normalTexture],
+        ["roughness", options.roughnessTexture]
+    ]);

```

Comment:
Any reason why we are using an actual Map instead of our classical JS object?

textureMap = { pattern: options.baseTexture, ....}

Also, seems weird but I have always read objects have faster lookup and are preferred over maps:

https://stackoverflow.com/questions/66931535/javascript-object-vs-map-set-key-lookup-performance
![image](https://user-images.githubusercontent.com/16060539/210993436-05c70f38-d8e6-4f99-b7d3-89ee8eb4ab9d.png)

Reply:
You are right, using other languages equals doing stuff like this 😅 
Thanks for the heads up!